ci/stm32cube/Src/main.c

```diff
@@ -0,0 +1,104 @@
+/* USER CODE BEGIN Includes */
+#include <micro_ros_platformio.h>
+
+#include <rcl/error_handling.h>
+#include <rcl/rcl.h>
+#include <rclc/executor.h>
+#include <rclc/rclc.h>
+#include <stdio.h>
+
+#include <std_msgs/msg/int32.h>
+/* USER CODE END Includes */
+
+/* USER CODE BEGIN PD */
+#define RCCHECK(fn)                \
+  {                                \
+    rcl_ret_t temp_rc = fn;        \
+    if ((temp_rc != RCL_RET_OK)) { \
+      Error_Handler();             \
+    }                              \
+  }
+#define RCSOFTCHECK(fn)            \
+  {                                \
+    rcl_ret_t temp_rc = fn;        \
+    if ((temp_rc != RCL_RET_OK)) { \
+    }                              \
+  }
+/* USER CODE END PD */
+
+/* USER CODE BEGIN PV */
+static rcl_publisher_t publisher;
+static std_msgs__msg__Int32 msg;
+static rclc_executor_t executor;
+static rclc_support_t support;
+static rcl_allocator_t allocator;
+static rcl_node_t node;
+static rcl_timer_t timer;
+
+static uint8_t uart_rbuffer[2048];
+static uint8_t uart_tbuffer[2048];
+static struct DMAStream stream = {
+    .uart = &huart1,
+    .rbuffer_size = 2048,
+    .rbuffer = uart_rbuffer,
+    .tbuffer_size = 2048,
+    .tbuffer = uart_tbuffer,
+};
+/* USER CODE END PV */
+
+/* USER CODE BEGIN 0 */
+void timer_callback(rcl_timer_t* timer, int64_t last_call_time) {
+  RCLC_UNUSED(last_call_time);
+  if (timer != NULL) {
+    RCSOFTCHECK(rcl_publish(&publisher, &msg, NULL));
+    msg.data++;
+  }
+}
+
+void HAL_UART_TxCpltCallback(UART_HandleTypeDef* huart) {
+  if (huart == &huart1) {
+    uart_transfer_complete_callback(&stream);
+  }
+}

```

Comment:
Could we move this inside of the library?

Reply:
Implementing `HAL_UART_TxCpltCallback` inside the library would be problematic in case the user wants to use another UART port for something else. I don't see a clean way of doing this, any ideas?